quipucords/scanner/network/processing/cpu.py

```diff
@@ -74,3 +74,80 @@ class ProcessCpuBogomips(process.Processor):
     def process(output, dependencies=None):
         """Pass the output back through."""
         return get_line(output['stdout_lines'])
+
+
+class ProcessCpuSocketCount(process.Processor):
+    """Process the cpu socket count."""
+
+    KEY = 'cpu_socket_count'
+    DEPS = ['internal_cpu_socket_count_dmi_cmd',
+            'internal_cpu_socket_count_cpuinfo_cmd',
+            'cpu_count']
+    REQUIRE_DEPS = False
+
+    @staticmethod
+    def process(output, dependencies):
+        """Return the correct value for cpu_socket_count output."""
+        # process the internal dmi cpu socket count result
+        dmi_cpu_socket_count = \
+            dependencies.get('internal_cpu_socket_count_dmi_cmd')
+        if dmi_cpu_socket_count and dmi_cpu_socket_count.get('rc') == 0:
+            dmi_count = dmi_cpu_socket_count.get('stdout').strip()
+            try:
+                if int(dmi_count) != 0:
+                    return dmi_count
+            except ValueError:
+                pass
+
+        # process the cpuinfo socket count as a fallback
+        cpuinfo_cpu_socket_count = \
+            dependencies.get('internal_cpu_socket_count_cpuinfo_cmd')
+        if cpuinfo_cpu_socket_count and \
+                cpuinfo_cpu_socket_count.get('rc') == 0 and \
+                cpuinfo_cpu_socket_count.get('stdout_lines'):
+            cpuinfo_count = cpuinfo_cpu_socket_count.get(
+                'stdout_lines', [0])[0]
+            try:
+                if int(cpuinfo_count) != 0:
+                    return cpuinfo_count
+            except ValueError:
+                pass
+
+        # assign the socket_count to the cpu_count as a last resort
+        cpu_count = dependencies.get('cpu_count')
+        return cpu_count
+
```

Comment:
You did an excellent job rewriting this. It is a lot easier to understand and is so clean. 🏅